@stellar/freighter-api/src/signAuthEntry.ts

```diff
@@ -8,13 +12,22 @@ export const signAuthEntry = async (
   opts?: {
     networkPassphrase?: string;
     address?: string;
-  }
+  },
 ): Promise<
   { signedAuthEntry: Buffer | null; signerAddress: string } & {
     error?: FreighterApiError;
   }
 > => {
   if (isBrowser) {
+    const { isAllowed } = await requestAllowedStatus();
```

Comment:
at the freighter-api level, if the current public key is not connected to the dapp, give the user an opportunity to connect